allennlp/models/bert_for_classification.py

```diff
@@ -0,0 +1,141 @@
+from typing import Dict, Union
+
+from overrides import overrides
+import torch
+from pytorch_pretrained_bert.modeling import BertModel
+
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.models.model import Model
+from allennlp.nn.initializers import InitializerApplicator
+from allennlp.training.metrics import CategoricalAccuracy
+
+
+@Model.register("bert_for_classification")
+class BertForClassification(Model):
+    """
+    An AllenNLP Model that runs pretrained BERT,
+    takes the pooled output, and adds a Linear layer on top.
+    If you want an easy way to use BERT for classification, this is it.
+    Note that this is a somewhat non-AllenNLP-ish model architecture,
+    in the sense that it goes straight from indexed tokens to pooled output
+    without using a ``TextFieldEmbedder`` or ``TokenEmbedder``; and as such it
+    implicitly assumes that you used the bert-pretrained token indexer.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    bert_model : ``Union[str, BertModel]``
+        The BERT model to be wrapped. If a string is provided, we will call
+        ``BertModel.from_pretrained(bert_model)`` and use the result.
+    num_labels : ``int``, optional (default: None)
+        How many output classes to predict. If not provided, we'll use the
+        vocab_size for the ``label_namespace``.
+    index : ``str``, optional (default: "bert")
+        The index of the token indexer that generates the BERT indices.
+    label_namespace : ``str``, optional (default : "labels")
+        Used to determine the number of classes if ``num_labels`` is not supplied.
+    trainable : ``bool``, optional (default : True)
+        If True, the weights of the pretrained BERT model will be updated during training.
+        Otherwise, they will be frozen and only the final linear layer will be trained.
+    initializer : ``InitializerApplicator``, optional
+        If provided, will be used to initialize the final linear layer *only*.
+    """
+    def __init__(self,
+                 vocab: Vocabulary,
+                 bert_model: Union[str, BertModel],
+                 num_labels: int = None,
+                 index: str = "bert",
+                 label_namespace: str = "labels",
+                 trainable: bool = True,
+                 initializer: InitializerApplicator = InitializerApplicator()) -> None:
+        super().__init__(vocab)
+
+        if isinstance(bert_model, str):
+            self.bert_model = BertModel.from_pretrained(bert_model)
```

Comment:
Unrelated to this PR, but this made me worried about coercing a json dict to a str in the `from_params` code.  Looks like we don't coerce, so this should be fine, but we also don't do explicit type checking, which it might be worth adding.